src/main/java/de/dentrassi/rpm/builder/RpmMojo.java

```diff
@@ -1396,4 +1448,54 @@ private static void ifSet ( final Consumer<String> setter, final String value, f
         }
     }
 
+    /**
+     * Read properties from the active profiles.
+     *
+     * Goes through all active profiles (in the order the
+     * profiles are defined in settings.xml) and extracts
+     * the desired properties (if present). The prefix is
+     * used when looking up properties in the profile but
+     * not in the returned map.
+     *
+     * @param prefix The prefix to use or null if no prefix should be used
+     * @param properties The properties to read
+     *
+     * @return A map containing the values for the properties that were found
+     */
+    public Map<String, String> readPropertiesFromActiveProfiles(final String prefix,
+                                                                 final String... properties) {
+        if (settings == null) {
+            this.logger.debug("No maven setting injected");
+            return Collections.emptyMap();
+        }
+
+        final List<String> activeProfilesList = settings.getActiveProfiles();
+        if (activeProfilesList.isEmpty()) {
+            this.logger.debug("No active profiles found");
+            return Collections.emptyMap();
+        }
+
+        final Map<String, String> map = new HashMap<String, String>();
```

Comment:
IIRC we are on Java 8. So you can drop the type arguments on the constructor:
```suggestion
        final Map<String, String> map = new HashMap<>();
```